<launch>
  <!-- Arguments for WhyCon node -->
  <arg name="name" default="whycon"/>
  <arg name="targets" default="1"/>
  <arg name="outer_diameter" default="0.117"/>
  <arg name="inner_diameter" default="0.48"/>
  <!-- <arg name="outer_diameter" default="5.5"/>
  <arg name="inner_diameter" default="2.0"/> -->
<!-- <arg name="outer_diameter" default="0.055"/>
  <arg name="inner_diameter" default="0.020"/> -->
  <!-- Arguments for ArUco node -->
  <!-- <arg name="markerSize"      default="0.15"/>   -->
    <!-- in m -->
  <!-- <arg name="side"             default="left"/> -->
  <!-- <arg name="ref_frame"       default=""/>  -->
   <!-- leave empty and the pose will be published wrt param parent_name -->

  <!-- Image_proc node for calibration of image to remove fish eye effect (required actually when we are using real camera) -->

<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="1280" />
    <param name="image_height" value="1024" />
    <!-- <param name="pixel_format" value="yuyv" /> -->
    <!-- <param name="camera_frame_id" value="usb_cam" /> -->
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <!-- <param name="framerate" value="30" /> -->
    <param name="io_method" value="mmap"/>
  </node>

   <group ns="usb_cam">
    <node pkg="image_proc" type="image_proc" name="image_proc"/>
  </group>

  <!-- Running WhyCon node-->
  <node name="whycon" type="whycon" pkg="whycon" output="screen">

    <!-- Setting the arguments recieved to the corresponding parameters -->
    <param name="targets" value="$(arg targets)"/>
    <param name="name" value="$(arg name)"/>
    <param name="outer_diameter" value="$(arg outer_diameter)"/>
    <param name="inner_diameter" value="$(arg inner_diameter)"/> 

    <!-- Remap the topics to the corresponding topics -->
    <remap from="/camera/camera_info" to="/usb_cam/camera_info"/>
    <remap from="/camera/image_rect_color" to="/usb_cam/image_rect_color"/>

  </node>


  <!-- Running WhyCon node-->
  <!-- <node name="whycon_raw" type="whycon" pkg="whycon" output="screen"> -->

    <!-- Setting the arguments recieved to the corresponding parameters -->
    <!-- <param name="targets" value="$(arg targets)"/> -->
    <!-- <param name="name" value="$(arg name)"/> -->
    <!-- <param name="outer_diameter" value="$(arg outer_diameter)"/> -->
    <!-- <param name="inner_diameter" value="$(arg inner_diameter)"/>  -->

    <!-- Remap the topics to the corresponding topics -->
    <!-- <remap from="/camera/camera_info" to="/usb_cam/camera_info"/> -->
    <!-- <remap from="/camera/image_rect_color" to="/usb_cam/image_raw"/> -->

  <!-- </node> -->


  <!-- Running image_view node to display WhyCon output -->
  <node ns = "whycon_display" name="image_view" type="image_view" pkg="image_view" output="screen">
    <!-- <remap from="image" to="/usb_cam/image_rect_color"/> -->
    <remap from="image" to="/whycon/image_out"/>
  </node>
</launch>